tests/components/azure_event_hub/test_init.py

```diff
@@ -90,16 +76,94 @@ async def test_full_config(hass):
             },
         }
     }
-    config[azure_event_hub.DOMAIN].update(MIN_CONFIG)
     assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
 
 
+async def test_setup(hass, mock_hub):
+    """Test the async_setup function."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    assert await azure_event_hub.async_setup_entry(hass, entry)
+    assert hass.data[azure_event_hub.DOMAIN]["hub"] is not None
+    assert isinstance(hass.data[azure_event_hub.DOMAIN]["hub"], AzureEventHub)
+
+
+async def test_unload_entry(hass, mock_hub):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    assert await azure_event_hub.async_setup_entry(hass, entry)
+    assert hass.data[azure_event_hub.DOMAIN].get("hub") is not None
+    assert await azure_event_hub.async_unload_entry(hass, entry)
+    assert hass.data[azure_event_hub.DOMAIN].get("hub") is None
+
+
+async def test_failed_test_connection(hass):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    with patch(
+        f"{PRODUCER_PATH}.get_eventhub_properties",
+        side_effect=EventHubError("test"),
+    ):
+        try:
+            await azure_event_hub.async_setup_entry(hass, entry)
```

Comment:
Always use core interfaces to start the tests. Here use `hass.config_entries.async_setup`. Then assert the state of the config entry.

https://github.com/home-assistant/core/blob/54d55fdf8804a22cf20276ab1d0c36978b51dcd3/homeassistant/config_entries.py#L90